### PR TITLE
Add test for user creation and deletion

### PR DESCRIPTION
**Pull request by olga-nesterova**

### CODE REVIEW

File: lib/testing_appserver/modules/people/user_creation_form.rb [hunk 1]
```diff
@@ -35,15 +35,15 @@ def wait_to_load
       @instance.webdriver.wait_until { first_name_element.present? }
     end
 
-    def create_new_user(params = {}, generate_pwd: false)
+    def create_new_user(params = {}, generate_pwd = false)
```
Comment (ShockwaveNN) [inline lib/testing_appserver/modules/people/user_creation_form.rb:38]: `rubocop` don't like it and I think it's a bad idea to have `params` and have `generate_pwd` key after that

I think `generate_pwd` should be part of `params`

File: spec/functional_tests/people/create_person_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 require 'spec_helper'
 
 user = TestingAppServer::UserData.new(first_name: 'John', last_name: 'Smith', mail: TestingAppServer::UserData::USER_EMAIL,
-                                      pwd: TestingAppServer::UserData::DEFAULT_USER_PASSWORD, type: :user)
+                                      pwd: SecureRandom.hex(5), type: :user)
```
Comment (ShockwaveNN) [inline spec/functional_tests/people/create_person_spec.rb:6]: I think we should make it a little longer that 5

File: lib/testing_appserver/data/user_data.rb [hunk 3]
```diff
@@ -8,8 +8,10 @@ class UserData
     DEFAULT_PORTAL = 'http://appserver.qa-onlyoffice.net/'
     DEFAULT_ADMIN_NAME = 'Administrator'
     DEFAULT_ADMIN_LASTNAME = ''
+    DEFAULT_USER_PASSWORD = '1234qwer'
```
Comment (ShockwaveNN) [inline lib/testing_appserver/data/user_data.rb:11]: Need to be removed

File: lib/testing_appserver/data/user_data.rb [hunk 4]
```diff
@@ -8,8 +8,10 @@ class UserData
...
+    USER_EMAIL = 'test@qamail.teamlab.info'
```
Comment (ShockwaveNN) [inline lib/testing_appserver/data/user_data.rb:12]: I think it should be cahnged to new created account for qamail

File: lib/testing_appserver/data/user_data.rb [hunk 5]
```diff
@@ -18,7 +20,9 @@ def initialize(user_data = {})
       @pwd =  user_data.fetch(:pwd, secret_data['portal_password_admin'])
       @first_name = user_data.fetch(:first_name, DEFAULT_ADMIN_NAME)
       @last_name = user_data.fetch(:last_name, DEFAULT_ADMIN_LASTNAME)
+      @full_name = "#{@first_name} #{@last_name}"
       @type = user_data.fetch(:type, :admin)
+      @pwd_init = user_data.fetch(:pwd_init, :temporary_pwd)
```
Comment (ShockwaveNN) [inline lib/testing_appserver/data/user_data.rb:25]: I don't like name of this parameter `pwd_init`

Maybe `pwd_generation_type` - better to understand

File: lib/testing_appserver/data/user_data.rb [hunk 6]
```diff
@@ -8,8 +8,10 @@ class UserData
     DEFAULT_PORTAL = 'http://appserver.qa-onlyoffice.net/'
     DEFAULT_ADMIN_NAME = 'Administrator'
     DEFAULT_ADMIN_LASTNAME = ''
+    DEFAULT_USER_PASSWORD = '1234qwer'
+    USER_EMAIL = 'test@qamail.teamlab.info'
 
-    attr_accessor :portal, :mail, :pwd, :first_name, :last_name, :type
+    attr_accessor :portal, :mail, :pwd, :first_name, :last_name, :type, :pwd_init, :full_name
```
Comment (ShockwaveNN) [inline lib/testing_appserver/data/user_data.rb:14]: I think `full_name` can be `attr_reader` - we don't need to change it usually

File: lib/testing_appserver/modules/people/user_creation_form.rb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+    def create_new_user(params = {}, generate_pwd = false)
+      fill_user_form(params, generate_pwd)
+      @instance.webdriver.move_to_element_by_locator(save_form_element.selector[:xpath])
+      save_form_element.click
+      @instance.webdriver.wait_until { success_toast_element.present? }
+      UserProfile.new(@instance)
+    end
+
+    def fill_user_form(params = {}, generate_pwd = false)
```
Comment (ShockwaveNN) [inline lib/testing_appserver/modules/people/user_creation_form.rb:46]: I think both those method can be private until we found any reason to make them public

**Review by ShockwaveNN — CHANGES_REQUESTED** [review]

File: lib/testing_appserver/data/user_data.rb (hunk 5 above)
Comment (olga-nesterova) [reply lib/testing_appserver/data/user_data.rb]: So there are gonna be two parameters which are very similar: `pwd_generation_type` and `generate_pwd`
I decided to rename it to `pwd_set_up_type`

File: .rubocop_todo.yml [hunk 8]
```diff
@@ -22,7 +22,14 @@ Style/MixinUsage:
   Exclude:
     - 'lib/testing_appserver/appserver_test_instance.rb'
 
-# Offense count: 1
+# Offense count: 2
+# Configuration parameters: AllowedMethods.
+# AllowedMethods: respond_to_missing?
+Style/OptionalBooleanParameter:
+  Exclude:
+    - 'lib/testing_appserver/modules/people/user_creation_form.rb'
```
Comment (ShockwaveNN) [inline .rubocop_todo.yml:30]: This is strange, is this should be fixed?

File: spec/functional_tests/people/create_person_spec.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+user = TestingAppServer::UserData.new(first_name: 'John', last_name: 'Smith',
+                                      mail: TestingAppServer::PrivateData.new.decrypt['user_mail'],
+                                      pwd: SecureRandom.hex(7), type: :user)
```
Comment (ShockwaveNN) [inline spec/functional_tests/people/create_person_spec.rb:7]: I would setup like `14` and it's enought

**Review by ShockwaveNN — APPROVED** [review]
You can fix two my notes and merge

File: spec/functional_tests/people/create_person_spec.rb (hunk 9 above)
Comment (olga-nesterova) [reply spec/functional_tests/people/create_person_spec.rb:7]: The resulting hexadecimal string is twice `n` (https://apidock.com/ruby/SecureRandom/hex/class), so now we are already generating 14 digits

**Pull request merged**